method-info/thousand-pound-correction/thousand_pound_correction.md

```diff
@@ -0,0 +1,153 @@
+    precision = 2
+)
+```
+The output will be returned as an object. You will need to destructure the object to extract the values and one way of doing this is using the built-in function vars(). vars() is used to return the \_\_dict\_\_ attribute for the specified module, class, instance or any other object with a \_\_dict\_\_ attribute.
+```python
+print(vars(output))
+```
+
+```bash
+{'unique_identifier': '12340000001-201409-q100', 'principal_final_value': '5.0E+4', 'target_variables': [TargetVariable(identifier='q101', original_value='500', final_value='0.5'), TargetVariable(identifier='q102', original_value='1000', final_value='1'), TargetVariable(identifier='q103', original_value='1500', final_value='1.5'), TargetVariable(identifier='q104', original_value=None, final_value=None)], 'tpc_ratio': '8.3E+2', 'tpc_marker': 'C'}
+
+```
+
+Example output table:
+
+| unique_identifier | principal_final_value | target_variables | tpc_ratio | tpc_marker |
+| --- | --- | --- | --- | --- |
+| '12340000001-201409-q100' | '5.0E+4' | [TargetVariable(identifier='q101', original_value='500', final_value='0.5'), TargetVariable(identifier='q102', original_value='1000', final_value='1'), TargetVariable(identifier='q103', original_value='1500', final_value='1.5'), TargetVariable(identifier='q104', original_value=None, final_value=None)] | '8.3E+2' | 'C' |
+
+Output attributes:
+* unique_identifier - Unique identifier e.g. a question code/ruref/period/id/combination of all of these
+* principal_final_value – Output value that may or may not be adjusted
+* target_variables - adjusted identifier/value pairs (returned as TargetVariable objects)
+* tpc_ratio – Ratio of the principal variable against good/predictive/aux response
+* tpc_marker - 'C' for correction applied, 'N' for no correction applied, 'S' for method stop / error
+An example output CSV file has been provided in the example_data folder.
+
+## Pandas Wrapper
+
+To view the code of the pandas wrapper you can find the `pandas_wrapper.py` file within the `utils` directory.
+
+### Prerequisites: Pandas Wrapper
+
+In order to run some of the functions in the python `pandas_wrapper.py`, you will need to have `pandas` and `numpy` installed.
+
+To install `pandas`:
+
+```python
+pip install pandas
+```
+
+To install `numpy`:
+
+```python
+pip install numpy
+```
+
+## Pandas Wrapper Usage
+
+- You will have to create a new python file importing in the `pandas_wrapper.py`.
+- Where you will have to write functions to read a CSV file and pass in the data as a DataFrame into the *`wrapper`* function from the `pandas_wrapper.py` file.
+
+
+We have an example of how to do this in the `pandas_example.py` file within the `utils` directory.
+
+
+### Additional Examples
+
+For more examples of using this method with CSV files or by implementing the pandas package see the totals_and_components source code [documentation page](https://github.com/ONSdigital/sml-python-small/blob/main/sml_small/editing/thousand_pounds).
```

Comment:
I think you forgot to replace `totals_and_components` source code with `thousand_pounds` source code =)

Reply:
Corrected now

Reply:
Thanks =) All good now!